web/workflow.R

```diff
@@ -39,29 +39,36 @@ if (is.na(args[1])){
   settings <- read.settings(settings.file)
 }
 
+# Update/fix/check settings. Will only run the first time it's called, unless force=TRUE
+settings <- fix.deprecated.settings(settings, force=FALSE)
+settings <- addSecrets(settings, force=FALSE)
+settings <- update.settings(settings, force=FALSE)
+settings <- check.settings(settings, force=FALSE)
+
 # Check for additional modules that will require adding settings
 if("benchmark" %in% names(settings)){
   library(PEcAn.benchmark)
   settings <- papply(settings, read.settings.RR)
 }
 
 # Write pecan.CHECKED.xml
-settings <- write.settings(settings, outputfile = "pecan.CHECKED.xml")
+write.settings(settings, outputfile = "pecan.CHECKED.xml")
```

Comment:
clean.settings is already in use.

I agree these functions are just a one-time sanitation process. Recall that benchmark_reference_runs stores a subset of a pecan xml settings object in the database, and those bits will definitely need cleaning/updating and localization to the specific machine they're running on.

I guess I like the idea of keeping all the parts of settings sanitation in one function because the average high-level user doesn't need to know/care about the details. In terms of how/whether those parts of sanitation are broken out into 1, 4, or n functions internally, I'm pretty agnostic